File: src/2025-03-04-03-28-40--deepseek-chat-0304/go/exercises/practice/wordy/wordy.go
package wordy

import (
	"strconv"
	"strings"
)

func Answer(question string) (int, bool) {
	// Check if question starts with "What is" and ends with "?"
	if !strings.HasPrefix(question, "What is ") || !strings.HasSuffix(question, "?") {
		return 0, false
	}

	// Remove prefix and suffix
	expr := strings.TrimPrefix(question, "What is ")
	expr = strings.TrimSuffix(expr, "?")

	// Split into tokens
	tokens := strings.Fields(expr)
	if len(tokens) == 0 {
		return 0, false
	}

	// Parse first number
	first, err := strconv.Atoi(tokens[0])
	if err != nil {
		return 0, false
	}

	result := first
	tokens = tokens[1:]

	for len(tokens) > 0 {
		if len(tokens) < 2 {
			return 0, false
		}

		// Get operation
		op := tokens[0]
		// Handle "multiplied by" and "divided by" as two tokens
		if len(tokens) > 1 && (op == "multiplied" || op == "divided") {
			if tokens[1] != "by" {
				return 0, false
			}
			// Skip "by" token
			tokens = tokens[1:]
		}
		// Get next number
		numStr := tokens[1]
		num, err := strconv.Atoi(numStr)
		if err != nil {
			return 0, false
		}

		// Perform operation
		switch op {
		case "plus":
			result += num
		case "minus":
			result -= num
		case "multiplied":
			result *= num
		case "divided":
			if num == 0 {
				return 0, false
			}
			result /= num
		default:
			return 0, false
		}

		tokens = tokens[2:]
	}

	return result, true
}

File: src/python-2025-02-23-yh-deepclaude-retry/python/exercises/practice/wordy/wordy.py
def answer(question):
    if not question.startswith("What is ") or not question.endswith("?"):
        raise ValueError("syntax error")
    expr = question[8:-1].strip()
    if not expr:
        raise ValueError("syntax error")
    
    words = expr.split()
    tokens = []
    i = 0
    while i < len(words):
        current_word = words[i]
        if current_word in ('multiplied', 'divided'):
            if i + 1 >= len(words) or words[i+1] != 'by':
                raise ValueError("syntax error")
            tokens.append('*' if current_word == 'multiplied' else '/')
            i += 2
        elif current_word == 'plus':
            tokens.append('+')
            i += 1
        elif current_word == 'minus':
            tokens.append('-')
            i += 1
        else:
            try:
                tokens.append(int(current_word))
                i += 1
            except ValueError:
                if current_word == 'by':
                    raise ValueError("syntax error")
                raise ValueError("unknown operation")
    
    if not tokens or len(tokens) % 2 == 0 or not isinstance(tokens[0], int):
        raise ValueError("syntax error")
    for idx in range(1, len(tokens), 2):
        if tokens[idx] not in '+-*/' or (idx + 1 >= len(tokens)) or not isinstance(tokens[idx+1], int):
            raise ValueError("syntax error")
    
    result = tokens[0]
    for idx in range(1, len(tokens), 2):
        op = tokens[idx]
        num = tokens[idx+1]
        if op == '+':
            result += num
        elif op == '-':
            result -= num
        elif op == '*':
            result *= num
        elif op == '/':
            result = result // num
    
    return result
